storm-core/src/jvm/backtype/storm/utils/Utils.java

```diff
@@ -533,4 +533,16 @@ private static SerializationDelegate getSerializationDelegate(Map stormConf) {
         delegate.prepare(stormConf);
         return delegate;
     }
+
+    public static void handleUncaughtException(Throwable t) {
+      if(t!= null && t instanceof OutOfMemoryError) {
+         try {
+             System.err.println("Halting due to Out Of Memory Error..." + Thread.currentThread().getName());
+         } catch (Throwable err) {
+             //Again we done want to exit because of logging issues.
+         }
+         Runtime.getRuntime().halt(-1);
+      }
+    }
```

Comment:
@lazyval, using log interface could rethrow OOM which won't be handled properly. Hence using `System.err.println`